app/main/controllers/media/RTMediaMeta.php

```diff
@@ -72,7 +72,7 @@ public function update_meta( $id = false, $key = false, $value = false, $duplica
 		if ( true === $duplicate ) {
 			$media_meta = $this->model->insert( array( 'media_id' => $id, 'meta_key' => $key, 'meta_value' => $value ) );
 		} else {
-			if ( $this->get_single_meta( $id, $key ) ) {
+			if ( false !== $this->get_single_meta( $id, $key ) ) {
```

Comment:
So for privacy `logged-in-users`, the meta value is empty?

Reply:
No, the privacy for `public` in 0.
When we change it to something else, it goes for checking, if a meta entry already exists. So, for `public` privacy, the entry exists which returns 0 as the meta value.
Now, this 0 is checked against `if` condition (for insert or updating the entry)
When we check 0 inside `if`, i.e., `if( 0 )`
This results in a falsy condition check which creates another row for `logged-in-users` privacy moderation.
Ideally, it should update the same entry.